modules/azure/function_app_linux_managed_identity/main.tf

```
@@ -50,12 +50,35 @@ resource "azurerm_linux_function_app" "function_app" {
   storage_account_access_key  = var.storage_account_access_key
   functions_extension_version = var.runtime_version
 
-  app_settings = merge(var.app_settings, { MICROSOFT_PROVIDER_AUTHENTICATION_SECRET = "${local.should_create_app ? azuread_application_password.password[0].value : var.managed_identity_provider.existing.client_secret}" })
+  app_settings = merge(var.app_settings, {
+    MICROSOFT_PROVIDER_AUTHENTICATION_SECRET = "${local.should_create_app ? azuread_application_password.password[0].value : var.managed_identity_provider.existing.client_secret}"
+  })
 
   site_config {
     always_on              = var.always_on
     vnet_route_all_enabled = var.route_all_outbound_traffic
-    ip_restriction         = var.ip_restriction
+
+    dynamic "ip_restriction" {
+      for_each = var.ip_restriction
```

```suggestion
      for_each = var.ip_restrictions
```
If we are changing this any way lets make this plural `ip_restrictions` like all the for_each's